File: frontend/src/types/projectMember.ts
/**
 * 项目成员类型定义
 * 对应后端 schemas/project_member.py 的 MemberOut / MemberAddIn / MemberRoleUpdateIn
 */

/** 项目内角色枚举（与后端 pattern ^(owner|reviewer)$ 对齐） */
export type ProjectRole = 'owner' | 'reviewer'

/** 项目成员响应项（对应后端 MemberOut） */
export interface ProjectMemberOut {
  /** 成员记录 ID */
  id: number
  /** 关联用户 ID */
  user_id: number
  /** 用户登录名 */
  username: string
  /** 用户昵称（可为空） */
  nickname?: string | null
  /** 项目内角色 */
  role_in_project: ProjectRole
  /** 加入时间 */
  create_time: string
}

/** 添加成员请求体（对应后端 MemberAddIn） */
export interface ProjectMemberAddIn {
  /** 被加入的用户 ID */
  user_id: number
  /** 项目内角色，默认 reviewer */
  role_in_project?: ProjectRole
}

/** 更新成员角色请求体（对应后端 MemberRoleUpdateIn） */
export interface ProjectMemberRoleUpdateIn {
  /** 新的项目内角色 */
  role_in_project: ProjectRole
}

File: backend/app/schemas/project_member.py
"""
项目成员 Pydantic Schema

v2.4: 用于 project_member 管理 API 的请求/响应
"""
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class MemberAddIn(BaseModel):
    """添加项目成员请求体"""
    user_id: int = Field(..., description="被加入的用户ID")
    role_in_project: str = Field(default="reviewer", pattern="^(owner|reviewer)$", description="项目内角色")


class MemberRoleUpdateIn(BaseModel):
    """更新成员角色请求体"""
    role_in_project: str = Field(..., pattern="^(owner|reviewer)$", description="新的项目内角色")


class MemberOut(BaseModel):
    """项目成员响应项

    R8 修复(2026-06-25):补齐 update_time,对齐 ProjectMember ORM。
    """
    id: int
    user_id: int
    username: str
    nickname: Optional[str] = None
    role_in_project: str
    create_time: datetime
    # R8 修复:补齐 update_time,对齐 ProjectMember ORM
    update_time: Optional[datetime] = None

    model_config = {"from_attributes": True}
